File: apps/backend-django/usuarios/dtos/invitacion.py
"""Datos que llegan al aceptar una invitación."""

from dataclasses import dataclass


@dataclass(frozen=True)
class AceptacionDeInvitacionDTO:
    """Lo que escribe el trabajador en el formulario del enlace.

    El correo y el rol no vienen de aquí: los puso el administrador al invitar
    y viajan dentro de la invitación. Si vinieran del formulario, cualquiera
    podría darse a sí mismo el rol de administrador.
    """

    token: str
    nombre: str
    apellido: str
    telefono: str
    contrasena: str

File: apps/frontend-react/src/modulos/autenticacion/dtos/invitacion.js
/** De qué es una invitación, para enseñarla antes de pedir los datos. */
export const invitacionPendienteDesdeApi = (invitacion) => ({
  correo: invitacion.correo,
  rol: invitacion.rol,
  negocio: invitacion.negocio,
  expiraEn: invitacion.expira_en,
})

/**
 * Lo que manda el formulario de la invitación. El correo y el rol no van: los
 * puso quien invitó, y el backend ignoraría lo que llegara.
 */
export const aceptacionHaciaApi = ({ token, nombre, apellido, telefono, contrasena }) => ({
  token,
  nombre: nombre.trim(),
  apellido: apellido.trim(),
  telefono: telefono.trim(),
  contrasena,
})
